File: week6/problem_set6/sentimental-readability/readability.py
# TODO

def main():
    text = input("Text: ")
    counters = counter(text)
    grade = readability_checker(counters)
    print_grade(grade)


def counter(text):
    alpha_count = sum(1 for char in text if char.isalpha())
    space_count = text.count(' ') + 1
    punctuation_count = sum(1 for char in text if char in '.?!')

    return [alpha_count, space_count, punctuation_count]


def readability_checker(counter):
    l = counter[0] / counter[1] * 100
    s = counter[2] / counter[1] * 100

    return round(0.0588 * l - 0.296 * s - 15.8)


def print_grade(grade):
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


main()

File: week2/problem_set2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int *counter(string text);
int readability_checker(int array[]);
void print_grade(int grade);
int main(void)
{
    string text = get_string("Text: ");

    int *counters = counter(text);

    int grade = readability_checker(counters);
    print_grade(grade);
}

int *counter(string text)
{
    int *counters = (int *) malloc(3 * sizeof(int));
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]))
            counters[0]++;
        else if (text[i] == ' ')
            counters[1]++;
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
            counters[2]++;
    }
    counters[1]++;

    return counters;
}

int readability_checker(int counters[])
{
    float l = (float) counters[0] / counters[1] * 100.0;
    float s = (float) counters[2] / counters[1] * 100.0;

    float index = 0.0588 * l - 0.296 * s - 15.8;

    return (int) (index + 0.5);
}

void print_grade(int grade)
{
    if (grade < 1)
        printf("Before Grade 1\n");
    else if (grade >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", grade);
}
